lib/puppet/type/cisco_bgp.rb

```diff
@@ -216,25 +216,41 @@ def self.title_patterns
     end
   end # property confederation_id
 
-  newproperty(:confederation_peers) do
+  newproperty(:confederation_peers, array_matching: :all) do
     desc "AS confederation parameters. Valid values are String,
-          keyword 'default'."
+          keyword 'default' or Array."
 
     match_error = 'must be specified in ASPLAIN or ASDOT notation'
+
     validate do |peers|
-      list = peers.split(' ')
-      list.each do |value|
+      if peers.is_a? String
+        peers_array = peers.split(' ')
+      else
+        peers_array = peers
+      end
+      peers_array.each do |value|
         fail "Confederation peer value '#{value}' #{match_error}" unless
-          /^\d+$/.match(value) || /^\d+\.\d+$/.match(value) ||
-          peers == 'default' || peers == :default
+          /^(\d+|\d+\.\d+)$/.match(value) ||
+          peers_array == 'default' || peers_array == :default
       end
+      peers
     end
 
     munge do |peers|
-      peers = :default if peers == 'default'
-      peers
+      if peers.is_a? String
+        if peers == 'default'
+          peers = :default
+        else
+          peers = peers.split(' ')
+        end
+      end
     end
-  end # property confederation_peers
+
+    def insync?(is)
+      (is.size == should.flatten.size && is.sort == should.flatten.sort) if is.any?
```

Comment:
This check will probably return false if is.any? is false (since it's the last command that processes in this method)